### ISSUE TITLE
[1.6.4] Crash with Rubidium 0.7.0/a

### ISSUE THREAD

**Issue by BaKYCoder** [Asek3/Oculus#459]
### Minecraft Version

Minecraft 1.20.1 [Forge 47.1.46]

### Oculus Version

oculus-mc1.20-1.6.4.jar

### Sodium Version

rubidium-mc1.20.1-0.7.0.jar

### Operating System

Windows 11

### What is your GPU?

NVIDIA GeForce GTX 1650 Max-Q Design

### Java Version

Java 17.0.3 (64bit)

### Reproduction Steps

1. Download **Rubidium** [0.7.0](https://www.curseforge.com/minecraft/mc-mods/rubidium/files/4683261) or [0.7.0a](https://www.curseforge.com/minecraft/mc-mods/rubidium/files/4684247).
2. Download **Oculus** [1.6.4](https://www.curseforge.com/minecraft/mc-mods/oculus/files/4578741).
3. Move the two mods to the **mods** folder.
4. Launch the **Minecraft Forge 1.20.1**.
5. The game **will crash** at the text: "_Loading bootstrap resources_".

### Crash Report file and latest.log

```shell
[18:20:49] [main/INFO]: ModLauncher running: args [--username, ******, --version, Forge 1.20.1, --gameDir, C:\Users\******\AppData\Roaming\.minecraft, --assetsDir, C:\Users\******\AppData\Roaming\.minecraft\assets, --assetIndex, 5, --uuid, *************************f700, --accessToken, ????????, --clientId, null, --xuid, null, --userType, mojang, --versionType, modified, --width, 925, --height, 530, --launchTarget, forgeclient, --fml.forgeVersion, 47.1.46, --fml.mcVersion, 1.20.1, --fml.forgeGroup, net.minecraftforge, --fml.mcpVersion, 20230612.114412]
[18:20:49] [main/INFO]: ModLauncher 10.0.9+10.0.9+main.dcd20f30 starting: java version 17.0.3 by Microsoft; OS Windows 10 arch amd64 version 10.0
[18:20:50] [main/INFO]: Loading ImmediateWindowProvider fmlearlywindow
[18:20:50] [main/INFO]: Trying GL version 4.6
[18:20:51] [main/INFO]: Requested GL version 4.6 got version 4.6
[18:20:52] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=union:/C:/Users/******/AppData/Roaming/.minecraft/libraries/org/spongepowered/mixin/0.8.5/mixin-0.8.5.jar%23100!/ Service=ModLauncher Env=CLIENT
[18:20:52] [pool-2-thread-1/INFO]: GL info: NVIDIA GeForce GTX 1650 with Max-Q Design/PCIe/SSE2 GL version 4.6.0 NVIDIA 536.40, NVIDIA Corporation
[18:20:52] [main/INFO]: Found mod file oculus-mc1.20-1.6.4.jar of type MOD with provider {mods folder locator at C:\Users\******\AppData\Roaming\.minecraft\mods}
[18:20:52] [main/INFO]: Found mod file rubidium-mc1.20.1-0.7.0.jar of type MOD with provider {mods folder locator at C:\Users\******\AppData\Roaming\.minecraft\mods}
[18:20:52] [main/WARN]: Mod file C:\Users\******\AppData\Roaming\.minecraft\libraries\net\minecraftforge\fmlcore\1.20.1-47.1.46\fmlcore-1.20.1-47.1.46.jar is missing mods.toml file
[18:20:52] [main/WARN]: Mod file C:\Users\******\AppData\Roaming\.minecraft\libraries\net\minecraftforge\javafmllanguage\1.20.1-47.1.46\javafmllanguage-1.20.1-47.1.46.jar is missing mods.toml file
[18:20:52] [main/WARN]: Mod file C:\Users\******\AppData\Roaming\.minecraft\libraries\net\minecraftforge\lowcodelanguage\1.20.1-47.1.46\lowcodelanguage-1.20.1-47.1.46.jar is missing mods.toml file
[18:20:52] [main/WARN]: Mod file C:\Users\******\AppData\Roaming\.minecraft\libraries\net\minecraftforge\mclanguage\1.20.1-47.1.46\mclanguage-1.20.1-47.1.46.jar is missing mods.toml file
[18:20:52] [main/INFO]: Found mod file fmlcore-1.20.1-47.1.46.jar of type LIBRARY with provider net.minecraftforge.fml.loading.moddiscovery.MinecraftLocator@5aae8eb5
[18:20:52] [main/INFO]: Found mod file javafmllanguage-1.20.1-47.1.46.jar of type LANGPROVIDER with provider net.minecraftforge.fml.loading.moddiscovery.MinecraftLocator@5aae8eb5
[18:20:52] [main/INFO]: Found mod file lowcodelanguage-1.20.1-47.1.46.jar of type LANGPROVIDER with provider net.minecraftforge.fml.loading.moddiscovery.MinecraftLocator@5aae8eb5
[18:20:52] [main/INFO]: Found mod file mclanguage-1.20.1-47.1.46.jar of type LANGPROVIDER with provider net.minecraftforge.fml.loading.moddiscovery.MinecraftLocator@5aae8eb5
[18:20:52] [main/INFO]: Found mod file client-1.20.1-20230612.114412-srg.jar of type MOD with provider net.minecraftforge.fml.loading.moddiscovery.MinecraftLocator@5aae8eb5
[18:20:52] [main/INFO]: Found mod file forge-1.20.1-47.1.46-universal.jar of type MOD with provider net.minecraftforge.fml.loading.moddiscovery.MinecraftLocator@5aae8eb5
[18:20:53] [main/INFO]: No dependencies to load found. Skipping!
[18:20:55] [main/INFO]: Compatibility level set to JAVA_17
[18:20:55] [main/INFO]: Successfully loaded Mixin Connector [org.tlauncher.MixinConnector]
[18:20:55] [main/INFO]: Launching target 'forgeclient' with arguments [--version, Forge 1.20.1, --gameDir, C:\Users\******\AppData\Roaming\.minecraft, --assetsDir, C:\Users\******\AppData\Roaming\.minecraft\assets, --uuid, *************************f700, --username, ******, --assetIndex, 5, --accessToken, ????????, --clientId, null, --xuid, null, --userType, mojang, --versionType, modified, --width, 925, --height, 530]
[18:20:55] [main/INFO]: Searching for graphics cards...
[18:20:56] [main/INFO]: Found graphics card: GraphicsAdapterInfo[vendor=NVIDIA, name=NVIDIA GeForce GTX 1650 with Max-Q Design, version=DriverVersion=31.0.15.3640]
[18:20:56] [main/INFO]: Found graphics card: GraphicsAdapterInfo[vendor=INTEL, name=Intel(R) UHD Graphics, version=DriverVersion=27.20.100.8190]
[18:20:56] [main/WARN]: One or more workarounds were enabled to prevent crashes or other issues on your system: [NVIDIA_THREADED_OPTIMIZATIONS]
[18:20:56] [main/WARN]: This might indicate that you need to update your graphics drivers.
[18:20:56] [main/WARN]: Mod 'oculus' attempted to override option 'mixin.features.sky', which doesn't exist, ignoring
[18:20:56] [main/WARN]: Mod 'oculus' attempted to override option 'mixin.features.entity.fast_render', which doesn't exist, ignoring
[18:20:56] [main/WARN]: Mod 'oculus' attempted to override option 'mixin.features.gui.font', which doesn't exist, ignoring
[18:20:56] [main/INFO]: Loaded configuration file for Rubidium: 40 options available, 0 override(s) found
[18:20:56] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/render/vertex/type/ChunkVertexBufferBuilder (java.lang.ClassNotFoundException: me.jellysquid.mods.sodium.client.render.vertex.type.ChunkVertexBufferBuilder)
[18:20:56] [main/WARN]: @Mixin target me.jellysquid.mods.sodium.client.render.vertex.type.ChunkVertexBufferBuilder was not found mixins.oculus.compat.sodium.json:block_id.MixinChunkVertexBufferBuilder
[18:20:56] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/render/chunk/tasks/ChunkRenderRebuildTask (java.lang.ClassNotFoundException: me.jellysquid.mods.sodium.client.render.chunk.tasks.ChunkRenderRebuildTask)
[18:20:56] [main/WARN]: @Mixin target me.jellysquid.mods.sodium.client.render.chunk.tasks.ChunkRenderRebuildTask was not found mixins.oculus.compat.sodium.json:block_id.MixinChunkRenderRebuildTask
[18:20:56] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/render/chunk/passes/BlockRenderPass (java.lang.ClassNotFoundException: me.jellysquid.mods.sodium.client.render.chunk.passes.BlockRenderPass)
[18:20:56] [main/WARN]: @Mixin target me.jellysquid.mods.sodium.client.render.chunk.passes.BlockRenderPass was not found mixins.oculus.compat.sodium.json:shader_overrides.MixinBlockRenderPass
[18:20:56] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/render/vertex/formats/ModelVertex (java.lang.ClassNotFoundException: me.jellysquid.mods.sodium.client.render.vertex.formats.ModelVertex)
[18:20:56] [main/WARN]: @Mixin target me.jellysquid.mods.sodium.client.render.vertex.formats.ModelVertex was not found mixins.oculus.compat.sodium.json:vertex_format.entity.MixinModelVertex
[18:20:56] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/render/chunk/format/ChunkMeshAttribute (java.lang.ClassNotFoundException: me.jellysquid.mods.sodium.client.render.chunk.format.ChunkMeshAttribute)
[18:20:56] [main/WARN]: @Mixin target me.jellysquid.mods.sodium.client.render.chunk.format.ChunkMeshAttribute was not found mixins.oculus.compat.sodium.json:vertex_format.ChunkMeshAttributeAccessor
[18:20:56] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/render/chunk/format/ChunkMeshAttribute (java.lang.ClassNotFoundException: me.jellysquid.mods.sodium.client.render.chunk.format.ChunkMeshAttribute)
[18:20:56] [main/WARN]: @Mixin target me.jellysquid.mods.sodium.client.render.chunk.format.ChunkMeshAttribute was not found mixins.oculus.compat.sodium.json:vertex_format.MixinChunkMeshAttribute
[18:20:56] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/render/vertex/type/ChunkVertexType (java.lang.ClassNotFoundException: me.jellysquid.mods.sodium.client.render.vertex.type.ChunkVertexType)
[18:20:56] [main/WARN]: @Mixin target me.jellysquid.mods.sodium.client.render.vertex.type.ChunkVertexType was not found mixins.oculus.compat.sodium.json:vertex_format.MixinModelVertexType
[18:20:57] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/render/chunk/region/RenderRegion$RenderRegionArenas (java.lang.ClassNotFoundException: me.jellysquid.mods.sodium.client.render.chunk.region.RenderRegion$RenderRegionArenas)
[18:20:57] [main/WARN]: @Mixin target me.jellysquid.mods.sodium.client.render.chunk.region.RenderRegion$RenderRegionArenas was not found mixins.oculus.compat.sodium.json:vertex_format.MixinRenderRegionArenas
[18:20:57] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/render/vertex/transform/CommonVertexElement (java.lang.ClassNotFoundException: me.jellysquid.mods.sodium.client.render.vertex.transform.CommonVertexElement)
[18:20:57] [main/WARN]: @Mixin target me.jellysquid.mods.sodium.client.render.vertex.transform.CommonVertexElement was not found mixins.oculus.compat.sodium.json:vertex_format.MixinCommonVertexElements
[18:20:57] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/render/vertex/transform/CommonVertexElement (java.lang.ClassNotFoundException: me.jellysquid.mods.sodium.client.render.vertex.transform.CommonVertexElement)
[18:20:57] [main/WARN]: @Mixin target me.jellysquid.mods.sodium.client.render.vertex.transform.CommonVertexElement was not found mixins.oculus.compat.sodium.json:vertex_format.CommonVertexElementAccessor
[18:20:57] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/render/vertex/serializers/VertexSerializerCache (java.lang.ClassNotFoundException: me.jellysquid.mods.sodium.client.render.vertex.serializers.VertexSerializerCache)
[18:20:57] [main/WARN]: @Mixin target me.jellysquid.mods.sodium.client.render.vertex.serializers.VertexSerializerCache was not found mixins.oculus.compat.sodium.json:vertex_format.MixinVertexSerializerCache
[18:20:57] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/render/vertex/transform/VertexTransform (java.lang.ClassNotFoundException: me.jellysquid.mods.sodium.client.render.vertex.transform.VertexTransform)
[18:20:57] [main/WARN]: @Mixin target me.jellysquid.mods.sodium.client.render.vertex.transform.VertexTransform was not found mixins.oculus.compat.sodium.json:vertex_format.MixinVertexTransform
[18:20:57] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/util/frustum/Frustum$Visibility (java.lang.ClassNotFoundException: me.jellysquid.mods.sodium.client.util.frustum.Frustum$Visibility)
[18:20:57] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/util/frustum/Frustum$Visibility (java.lang.ClassNotFoundException: me.jellysquid.mods.sodium.client.util.frustum.Frustum$Visibility)
[18:20:57] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/util/frustum/Frustum$Visibility (java.lang.ClassNotFoundException: me.jellysquid.mods.sodium.client.util.frustum.Frustum$Visibility)
[18:20:57] [main/WARN]: Error loading class: me/jellysquid/mods/sodium/client/util/frustum/Frustum$Visibility (java.lang.ClassNotFoundException: me.jellysquid.mods.sodium.client.util.frustum.Frustum$Visibility)
[18:21:03] [Datafixer Bootstrap/INFO]: 188 Datafixer optimizations took 236 milliseconds
```

**Comment by BaKYCoder** [comment]
If I download Rubidium [0.6.5 ](https://www.curseforge.com/minecraft/mc-mods/rubidium/files/4573226) and Oculus [1.6.4](https://www.curseforge.com/minecraft/mc-mods/oculus/files/4578741). the game will launch without problems.

**Comment by agentwolf88** [comment]
Same

**Comment by A00826925** [comment]
Bump, can confirm

**Comment by 301haru** [comment]
same, problem occurs in neoforge too

**Comment by MrKashew** [comment]
will this compatibility w new Rubidium be fixed in future - as FFAPI compatibility is fixed on 0.7.0a
https://github.com/Sinytra/ForgifiedFabricAPI/issues/22
